### PR TITLE
CDAP-3118 Add SparkSQL documentation

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-3118

### CODE REVIEW

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 1]
```diff
@@ -249,6 +249,24 @@ Spark in Workflows
 Spark programs in CDAP can also be added to a :ref:`workflow <workflows>`, similar to a :ref:`MapReduce <mapreduce>`.
 
 
+Spark SQL
+=========
+
+As you might know the `entry point into all functionality in Spark SQL
```
Comment (bdmogal) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: 'As you might know' may not be the best way to start this section. @awholegunch, any better suggestions?

**Comment by rohitsinha54** [comment]
Opened a jira for an example too: https://issues.cask.co/browse/CDAP-4843

**Comment by awholegunch** [comment]
@rohitsinha54 @bdmogal I have edited this. Please review. LGTM.

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 2]
```diff
@@ -249,6 +249,24 @@ Spark in Workflows
 Spark programs in CDAP can also be added to a :ref:`workflow <workflows>`, similar to a :ref:`MapReduce <mapreduce>`.
 
 
+Spark SQL
+=========
+The entry point to functionality in Spark SQL is through a Spark `SQLContext
+<http://spark.apache.org/docs/latest/sql-programming-guide.html#starting-point-sqlcontext>`__.
+To run a Spark SQL program in CDAP, you can obtain a ``SQLContext`` from CDAP's ``SparkContext``
+using either of these methods:
```
Comment (rohitsinha54) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: either of these sounds like they can use anyone among the below two but here they can only use one depending upon the language they are using. Do you think we can use some other word ? 

Besides the above comment LGTM.

Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: Sure. I will change it to "using one of these methods:"

**Comment by rohitsinha54** [comment]
LGTM

**Comment by rohitsinha54** [comment]
Docs updated to (temp): http://docs.cask.co/cdap/3.3.1-SNAPSHOT/en/developers-manual/building-blocks/spark-programs.html#spark-sql

**Pull request merged**